### ISSUE TITLE
Numerical issues in CLR / ILR

### ISSUE THREAD

**Issue by mortonjt** [biocore/scikit-bio#1861]
This is an issue that has been lurking in the background, but I have forgotten to raise

The CLR inverse transform has a numerical stability issue (since it is basically a softmax transform). Briefly, there can be overflow due to the exponentation operation.  The way around this is to use the [softmax trick](https://jamesmccaffrey.wordpress.com/2016/03/04/the-max-trick-when-computing-softmax/). Softmax is shift invariant (aka `clr_inv(x) = clr_inv(x-x.max(dim=-1))`), so won't need to exponentiate large numbers internally.

The ILR transform has similar issues -- right now we are representing the basis in probabilities, when we should be representing it in terms of logits.  This is what we are currently doing in gneiss -- but it would be nice to have all of this implemented in skbio for longer term maintenance 

- [x] Implement softmax trick in CLR inverse transform
- [x] Change ILR basis to logits instead of probabilities.

**Comment by mortonjt** [comment]
This has been addressed by #1862
